src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -21,19 +27,11 @@ namespace Microsoft.CodeAnalysis.Editor.InlineHints
     /// cref="InlineHintsTag"/>, which actually creates the UIElement. It reacts to tags changing and updates the
     /// adornments accordingly.
     /// </summary>
-    internal sealed class InlineHintsTagger : ITagger<IntraTextAdornmentTag>, IDisposable
+    internal sealed class InlineHintsTagger : EfficientTagger<IntraTextAdornmentTag>
     {
-        private readonly ITagAggregator<InlineHintDataTag> _tagAggregator;
+        private static ConditionalWeakTable<TagSpan<InlineHintDataTag>, TagSpan<IntraTextAdornmentTag>> s_dataTagToAdornmentTag = new();
```

Comment:
Note: if we're concerned about a CWT here, we could switch to a more explicit dictionary that we clear whenever the text snapshot changes.